### PR TITLE
Execute order

### PR DESCRIPTION
**Pull request by rogercloud**

### CODE REVIEW

File: curp/src/keybased_channel.rs [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    }
+}
+
+impl<K: Eq + Hash + Clone, M> KeybasedChannelReceiver<K, M> {
+    /// Receive a message
+    /// Return (message, `msg_complete_sender`)
+    #[allow(dead_code)]
+    pub(crate) fn recv(&self) -> (M, UnboundedSender<K>) {
+        loop {
+            let lisenter = {
```
Comment (pzheng1025) [inline curp/src/keybased_channel.rs:118]: possible to move the common logic to a helper function?

**Comment by pzheng1025** [comment]
Is this a MPSC channel?

File: curp/src/keybased_channel.rs (hunk 1 above)
Comment (rogercloud) [reply curp/src/keybased_channel.rs:118]: I've taken it into consideration, but there's logic to return the value in the middle of a loop, line 123. A function can't work that way.

**Comment by rogercloud** [comment]
> Is this a MPSC channel?

It's a MPMC channel, so it can be used as a MPSC channel.

**Review by pzheng1025 — APPROVED** [review]

**Pull request merged**